common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -899,9 +908,15 @@ def answer_available(self):
             return self.is_correct() or self.is_past_due()
         elif self.showanswer == SHOWANSWER.PAST_DUE:
             return self.is_past_due()
+        elif self.showanswer == SHOWANSWER.AFTER_ATTEMPTS:
+            # self.attempts_before_showanswer_button is read_only, to
+            # work with its modified value, we make a copy.
+            required_attempts = self.attempts_before_showanswer_button
+            if self.max_attempts and required_attempts >= self.max_attempts:
+                required_attempts = self.max_attempts
+            return self.attempts >= required_attempts
         elif self.showanswer == SHOWANSWER.ALWAYS:
             return True
-
```

Comment:
This is an unnecessary change, so we should revert it.

We want to keep PRs focused and as minimal as possible,
so we should only change lines that apply to the feature at hand.

Unrelated cleanup can be addressed in a separate PR.